src/test/spec/unified_runner/test_event.rs

```diff
@@ -1,38 +1,191 @@
-use crate::{bson::Document, test::CommandEvent};
+use crate::{
+    bson::Document,
+    event::cmap::{ConnectionCheckoutFailedReason, ConnectionClosedReason},
+    test::{CmapEvent, CommandEvent, Event},
+};
 use serde::Deserialize;
 
 #[derive(Debug, Deserialize, PartialEq)]
-pub enum TestEvent {
-    #[serde(rename = "commandStartedEvent")]
+#[serde(untagged, deny_unknown_fields, rename_all = "camelCase")]
+pub enum ExpectedEvent {
+    Cmap(ExpectedCmapEvent),
+    Command(ExpectedCommandEvent),
+    Sdam,
+}
+
+impl From<Event> for ExpectedEvent {
+    fn from(event: Event) -> Self {
+        match event {
+            Event::Command(sub) => ExpectedEvent::Command(sub.into()),
+            Event::Cmap(sub) => ExpectedEvent::Cmap(sub.into()),
+            Event::Sdam(_) => ExpectedEvent::Sdam,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub enum ExpectedCommandEvent {
+    #[serde(rename = "commandStartedEvent", rename_all = "camelCase")]
```

Comment:
To my surprise, `rename_all = "camelCase"` on an `enum` only applies to the names of the variants, not the fields of the individual variants.